server/services/v1/cache/runner.go

```diff
@@ -221,14 +220,30 @@ func (runner *SetRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*Cac
 	options := &cache.SetOptions{
 		NX: runner.req.GetNx(),
 		XX: runner.req.GetXx(),
-		EX: time.Duration(runner.req.GetEx()) * time.Second,
+		EX: runner.req.GetEx(),
+		PX: runner.req.GetPx(),
 	}
+
+	var oldVal *internal.CacheData
+	if runner.req.GetGet() {
+		oldVal, _ = runner.cacheStore.Get(ctx, tableName, runner.req.GetKey(), &cache.GetOptions{
```

Comment:
why not use `GetSet` API in store for this?

Reply:
GetSet doesn't accept expiration options.

Reply:
`setArgs` returns the old value but always in form of `string`.

Reply:
Then can we have GetSet as a separate API that doesn't accept expiration?

Reply:
sure https://github.com/tigrisdata/tigris-api/pull/188

Reply:
```
As of Redis version 6.2.0, this command is regarded as deprecated.
```

Reply:
updated the PR.